contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -654,28 +656,32 @@ contract BridgeL2SovereignChain is
     /**
      * @notice Burn tokens from wrapped token to execute the bridge, if the token is not mintable it will be transferred
      * note This function has been extracted to be able to override it by other contracts like Bridge2SovereignChain
-     * @param tokenWrapped Wrapped token to burnt
+     * @param tokenWrapped Proxied Wrapped token to burnt
      * @param amount Amount of tokens
      * @return Amount of tokens that must be added to the leaf after the bridge operation
      * @dev in case of tokens with non-standard transfers behavior like fee-on-transfer tokens or Max-value amount transfers user balance tokens,
      * It is possible that the amount of tokens sent is different from the amount of tokens received, in those cases, the amount that should be
      * added to the leaf has to be the amount received by the bridge
      */
     function _bridgeWrappedAsset(
-        TokenWrapped tokenWrapped,
+        ITokenWrappedBridgeUpgradeable tokenWrapped,
         uint256 amount
     ) internal override returns (uint256) {
         // The token is either (1) a correctly wrapped token from another network
         // or (2) wrapped with custom contract from origin network
         if (wrappedAddressIsNotMintable[address(tokenWrapped)]) {
-            uint256 balanceBefore = tokenWrapped.balanceOf(address(this));
+            // Swap interface from ITokenWrappedBridgeUpgradeable to IERC20Upgradeable for ERC20 functions access.
```

Comment:
If we also add the standard ERC20 interface into the `ITokenWrappedBridgeUpgradeable`, this is not needed.